README.markdown

```diff
@@ -195,72 +195,216 @@ int main(int argc, char**argv) {
+
+// g3log unit tests
+ADD_G3LOG_UNIT_TEST:BOOL=OFF
+
+// Use DBUG logging level instead of DEBUG.
+// By default DEBUG is the debugging level
+CHANGE_G3LOG_DEBUG_TO_DBUG:BOOL=OFF
+
+// Specifies the build type on single-configuration generators.
+// Possible values are empty, Debug, Release, RelWithDebInfo, MinSizeRel, … 
+CMAKE_BUILD_TYPE:STRING=
+
+// Install path prefix, prepended onto install directories.
+// This variable defaults to /usr/local on UNIX
+// and c:/Program Files/${PROJECT_NAME} on Windows.
+CMAKE_INSTALL_PREFIX:PATH=
+
+// The prefix used in the built package.
+// On Linux, if this option is not set:
+// 1) If CMAKE_INSTALL_PREFIX is given, then it will be
+//    set with the value of CMAKE_INSTALL_PREFIX by g3log.
+// 2) Otherwise, it will be set as /usr/local by g3log. 
+CPACK_PACKAGING_INSTALL_PREFIX:PATH=
+
+// Enable Visual Studio break point when receiving a fatal exception.
+// In __DEBUG mode only
+DEBUG_BREAK_AT_FATAL_SIGNAL:BOOL=OFF
+
+// Vectored exception / crash handling with improved stack trace
+ENABLE_FATAL_SIGNALHANDLING:BOOL=ON
+
+// Vectored exception / crash handling with improved stack trace
+ENABLE_VECTORED_EXCEPTIONHANDLING:BOOL=ON
+
+// iOS version of library.
+G3_IOS_LIB:BOOL=OFF
+
+// Log full filename
+G3_LOG_FULL_FILENAME:BOOL=OFF
+
+// Build shared library
+G3_SHARED_LIB:BOOL=ON
+
+// Build shared runtime library MSVC
+G3_SHARED_RUNTIME:BOOL=ON
+
+// Turn ON/OFF log levels.
+// An disabled level will not push logs of that level to the sink.
+// By default dynamic logging is disabled
+USE_DYNAMIC_LOGGING_LEVELS:BOOL=OFF
+
+// Use dynamic memory for message buffer during log capturing
+USE_G3_DYNAMIC_MAX_MESSAGE_SIZE:BOOL=OFF
+
+...
+```
+If you want leave everything as it was, then you should:
+```
+cmake ..
+```
+You may also specify one or more of those options listed above from the command line.
+For example, on Windows:
+```
+cmake .. -G "Visual Studio 15 2017" -DG3_SHARED_LIB=OFF -DCMAKE_INSTALL_PREFIX=C:/g3log -DADD_G3LOG_UNIT_TEST=ON -DADD_FATAL_EXAMPLE=OFF
+```
+will use a Visual Studio 2017 solution generator, build g3log as a static library, headers and libraries will be installed to `C:\g3log` when installed from source, enable unit testing, but do not build fatal example.
+
+*Note*: To build the tests, you should uncompress `g3log/3rdParty/gtest/gtest-1.7.0.zip` first. On Linux, you may:
+```
+# Suppose you are still in `g3log/build`
+pushd ../3rdParty/gtest
+unzip gtest-1.7.0.zip
+popd
+```
+
+MinGW users on Windows may find they should use a different generator:
+```
+cmake .. -G "MinGW Makefiles"
+```
+
+By default, headers and libraries will be installed to `/usr/local` on Linux when installed from source via `make install`. You may overwrite it by:
+```
+cmake .. -DCMAKE_INSTALL_PREFIX=/usr
+```
+This will install g3log to `/usr` instead of `/usr/local`.
 
+Package maintainers may be interested in the `CPACK_PACKAGING_INSTALL_PREFIX`. For example:

```

Comment:
nit:  Make it clear that this does not work on Windows. 
Ref: [your testing](https://github.com/myd7349/Ongoing-Study/blob/master/cpp/CMake/libfoo_v2/README.md#test-b30cd9360990fd439682e3543705aab7ea2d49fc) that shows that specifying `CPACK_PACKAGING_INSTALL_PREFIX` fails with `cpack`.

Suggestion:
`Package maintainers` 
-> `*nix* package maintainers` or 
-> `Linux/OSX package maintainers`

Reply:
Got it.